game-core/src/main/java/games/strategy/triplea/ui/screen/TileManager.java

```diff
@@ -336,24 +339,27 @@ private static void drawTerritoryEffects(final Territory territory, final MapDat
   }
 
   private void drawUnits(final Territory territory, final MapData mapData, final Set<Tile> drawnOn,
-      final Set<IDrawable> drawing) {
+      final Set<IDrawable> drawing, final GameData data) {
     final Iterator<Point> placementPoints = mapData.getPlacementPoints(territory).iterator();
     if (placementPoints == null || !placementPoints.hasNext()) {
       throw new IllegalStateException("No where to place units:" + territory.getName());
     }
+
     Point lastPlace = null;
-    final Iterator<UnitCategory> unitCategoryIter =
-        UnitSeperator.categorize(territory.getUnits().getUnits()).iterator();
-    while (unitCategoryIter.hasNext()) {
-      final UnitCategory category = unitCategoryIter.next();
+    final List<UnitCategory> categories = getSortedUnitCategories(territory, data);
+    for (final UnitCategory category : categories) {
       final boolean overflow;
       if (placementPoints.hasNext()) {
         lastPlace = new Point(placementPoints.next());
         overflow = false;
       } else {
         lastPlace = new Point(lastPlace);
-        lastPlace.x += uiContext.getUnitImageFactory().getUnitImageWidth();
         overflow = true;
+        if (mapData.getPlacementOverflowToLeft(territory)) {
+          lastPlace.x -= uiContext.getUnitImageFactory().getUnitImageWidth();
+        } else {
+          lastPlace.x += uiContext.getUnitImageFactory().getUnitImageWidth();
+        }
```

Comment:
I'm not sure if I'd prefer something like this:
```java
lastPlace.x += (mapData.getPlacementOverflowToLeft(territory) ? -1 :  1) * uiContext.getUnitImageFactory().getUnitImageWidth();
```
Not sure if this makes the code more readable, but at least this makes the difference between those 2 method calls obvious. I missed this detail when I first read the code

Reply:
Consider that. I felt that the if/else was easier to read and more intuitive in this case even if it is a bit subtle. I think its about a wash between the 2 approaches so I'll leave as is.

Reply:
I'm fine with it 👍